src/indexing/EventIndex.js

```diff
@@ -240,6 +240,38 @@ export default class EventIndex extends EventEmitter {
         this.crawlerCheckpoints.push(backwardsCheckpoint);
     }
 
+    /**
+     * Check if an event should be added to the event index.
+     *
+     * Most notably we filter events for which decryption failed, are redacted
+     * or aren't of a type that we know how to index.
+     *
+     * @param {MatrixEvent} ev The event that should checked.
+     * @returns {bool} Returns true if the event can be indexed, false
+     * otherwise.
+     */
+    isValidEvent(ev) {
+        const validEventType = ([
+            "m.room.message",
+            "m.room.name",
+            "m.room.topic",
+            ].indexOf(ev.getType()) >= 0
+                && !ev.isRedacted() && !ev.isDecryptionFailure()
+        );
```

Comment:
It's probably best to split this up into a couple statements for clarity, something like:

```js
const isUsefulType = ["m.room.message", "m.room.name", "m.room.topic"].includes(ev.getType());
const validEventType = isUsefulType && !ev.isRedacted() && !ev.isDecryptionFailure();
```